instrument-cloudwatch/src/Instrument/CloudWatch.hs

```diff
@@ -282,3 +289,23 @@ httpRetryH = const $ EX.Handler $ \(_ :: Amazonka.HttpException) -> return True
 -- | 'IOException's should be retried
 networkRetryH :: Monad m => a -> EX.Handler m Bool
 networkRetryH = const $ EX.Handler $ \(_ :: EX.IOException) -> return True
+
+-------------------------------------------------------------------------------
+newtype GzippedPutMetricData = GzippedPutMetricData CW.PutMetricData
+  deriving newtype (AWS.ToPath, AWS.ToQuery, AWS.ToHeaders)
+
+instance Amazonka.Types.AWSRequest GzippedPutMetricData where
```

Comment:
This should probably have a comment explaining what's happening and the rationale for only gzipping sometimes.

Reply:
added a description.